reposerver/repository/repository.go

```diff
@@ -1236,16 +1238,27 @@ func getResolvedValueFiles(
 		}
 
 		if !isRemote {
-			_, err = os.Stat(string(resolvedPath))
-			if os.IsNotExist(err) {
+			paths, err := doublestar.FilepathGlob(string(resolvedPath))
+
+			if err != nil {
+				return nil, fmt.Errorf("failed to get Helm values file by glob pattern: %w", err)
+			}
+			// To guarantee lexical order
+			sort.Strings(paths)
+			if len(paths) > 0 {
+				for _, path := range paths {
+					resolvedValueFiles = append(resolvedValueFiles, pathutil.ResolvedFilePath(path))
```

Comment:
I think this effectively allows the user to bypass symlink protection. Instead of returning `pathutil.ResolvedFilePath(path)`, sw should run `FilepathGlob` with `WithNoFollow()` and then run the output through `pathutil.ResolveValueFilePathOrUrl` (to make sure the path isn't to an out-of-bounds symlink).

Reply:
We should also have a test with values something like this:

```yaml
valueFiles
- *.yaml
```

Where `blah.yaml` is a symlink to `../sensitive.yaml`.

We should assert that this access attempt is blocked.

Reply:
Thanks, I 'll look into it monday probably

Reply:
I added 2 tests : 
- one with the out-of-bounds symlink in a referenced repo (using the ref variable) using 
- another with a single helm chart (using pre-existing out-of-bounds-values-file-link testdata)

Both of them are passing without any change in the code, repos are already checked for out-of-bounds symlink with argopath.CheckOutOfBoundsSymlinks beforehand.

Reply:
I'm uncertain about how to handle this case... `CheckOutOfBoundsSymlinks` is meant to be a "heavy-handed" check. I think many people disable it because their repo contains out-of-bounds symlinks which are not actually used by Argo CD. I kind of think we should still protect users here even if they've disabled the OOB symlink check.

But IMO it depends on how gnarly the implementation ends up being. Would you mind taking another look to see how rough it would be?

Reply:
Sorry for the delay on my side as well ;) (these last weeks have been quite busy for me)
I added a check for OOB symlinks for when AllowOutOfBoundsSymlinks is true.